### PR TITLE
feat: Allow relayer fee user to pass in token price to use

### PR DESCRIPTION
**Pull request by nicholaspai**
This will allow Vercel API for example to precompute Coingecko price and pass into SDK so that SDK doesn't have to make API call

### CODE REVIEW

**Review by kevinuma — APPROVED** [review]

File: src/relayFeeCalculator/chain-queries/arbitrum.ts [hunk 1]
```diff
@@ -27,9 +27,10 @@ export class ArbitrumQueries implements QueryInterface {
     return estimateTotalGasRequiredByUnsignedTransaction(tx, this.simulatedRelayerAddress, this.provider);
   }
 
-  async getTokenPrice(tokenSymbol: string): Promise<number> {
+  async getTokenPrice(tokenSymbol: string, coingeckoProApiKey?: string): Promise<number> {
```
Comment (kevinuma) [inline src/relayFeeCalculator/chain-queries/arbitrum.ts:30]: Can this key be passed in the relayerFeeCalculator config?

File: src/coingecko/Coingecko.ts [hunk 2]
```diff
@@ -33,7 +33,12 @@ export class Coingecko {
     return this.instance;
   }
 
-  private constructor(private readonly host = "https://api.coingecko.com/api/v3") {}
+  public static getPro(apiKey?: string) {
+    if (!this.instance) this.instance = new Coingecko("https://pro-api.coingecko.com/api/v3", apiKey);
+    return this.instance;
+  }
```
Comment (mrice32) [inline src/coingecko/Coingecko.ts:39]: nit: can we just drop the host param from the get method and combine get and getPro?

**Review by mrice32 — COMMENTED** [review]
Looks great! One small question

File: src/coingecko/Coingecko.ts (hunk 2 above)
Comment (nicholaspai) [reply src/coingecko/Coingecko.ts:39]: How would that work?

File: src/relayFeeCalculator/chain-queries/arbitrum.ts (hunk 1 above)
Comment (nicholaspai) [reply src/relayFeeCalculator/chain-queries/arbitrum.ts]: Even better, I can inject into the Query constructor

File: src/coingecko/Coingecko.ts (hunk 2 above)
Comment (nicholaspai) [reply src/coingecko/Coingecko.ts]: Like this? https://github.com/across-protocol/sdk-v2/pull/70/commits/6b6fa44da99abf43859950696a68d284c794af59

File: src/relayFeeCalculator/relayFeeCalculator.ts [hunk 3]
```diff
@@ -80,7 +84,9 @@ export class RelayFeeCalculator {
       console.error(`ERROR(gasFeePercent): Error while fetching token price ${error}`);
       throw error;
     });
-    const [gasCosts, tokenPrice] = await Promise.all([getGasCosts, getTokenPrice]);
+    const results = await Promise.all(_tokenPrice !== undefined ? [getGasCosts, getTokenPrice] : [getGasCosts]);
+    const gasCosts = results[0];
+    const tokenPrice = (_tokenPrice !== undefined ? _tokenPrice : results[1]) as number;
```
Comment (mrice32) [inline src/relayFeeCalculator/relayFeeCalculator.ts:89]: Promise.all works on non-promises, too!

```suggestion
    const [gasCosts, tokenPrice] = await Promise.all([getGasCosts, _tokenPrice !== undefined ? _tokenPrice : getTokenPrice]);
```

**Review by mrice32 — APPROVED** [review]
Looks great!

File: src/coingecko/Coingecko.ts (hunk 2 above)
Comment (mrice32) [reply src/coingecko/Coingecko.ts]: Yep, that's what I was thinking!

**Review by kevinuma — APPROVED** [review]

**Pull request merged**